compiler/src/main/java/org/qbicc/type/definition/LoadedTypeDefinition.java

```diff
@@ -210,25 +230,27 @@ default MethodElement resolveMethodElementVirtual(String name, MethodDescriptor
         // 2.b Otherwise, if C declares a method with the name and descriptor specified by
         // the method reference, method lookup succeeds.
 
-        int result = findMethodIndex(name, descriptor);
+        int result = findMethodIndex(name, descriptor, includePrivate);
         if (result != -1) {
             return getMethod(result);
         }
 
         // 2.c Otherwise, if C has a superclass, step 2 of method resolution is recursively
         // invoked on the direct superclass of C.
+        //
+        // We exclude private methods from this search because they cannot be inherited.
 
         LoadedTypeDefinition superClass = getSuperClass();
         if ( superClass != null ) {
-            MethodElement superCandidate = superClass.resolveMethodElementVirtual(name, descriptor);
+            MethodElement superCandidate = superClass.resolveMethodElementVirtual(name, descriptor, false);
             if ( superCandidate != null ) {
                 return superCandidate;
             }
         }
 
         int interfaceCount = getInterfaceCount();
         for (int i = 0; i < interfaceCount; i ++) {
-            MethodElement interfaceCandidate = getInterface(i).resolveMethodElementInterface(name, descriptor);
+            MethodElement interfaceCandidate = getInterface(i).resolveMethodElementInterface(true, name, descriptor);
             if (interfaceCandidate != null) {
                 return interfaceCandidate;
             }
```

Comment:
Just an FYI on this bit of code. With this change, it should no longer be necessary, since we've copied all interface methods down on to the class.  I've kept it here for now just because it is an easy place to set a breakpoint to see if it is still being triggered.  So as a note to myself, this should be deleted before the PR is ready for review.